### PR TITLE
Eliminate managed type new (mostly)

### PR DESCRIPTION
**Pull request by alyn509**
VM functionality for this not yet available on mainnet, safer not to merge yet.

### CODE REVIEW

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 1]
```diff
@@ -41,7 +41,7 @@ macro_rules! binary_operator {
             type Output = BigUint<M>;
 
             fn $method(self, other: u32) -> BigUint<M> {
-                let other_handle = M::managed_type_impl().bi_new(other as i64);
+                let other_handle = M::managed_type_impl().bi_new_handle(other as i64);
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:44]: We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 2]
```diff
@@ -51,7 +51,7 @@ macro_rules! binary_operator {
...
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:54]: Same.
We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 3]
```diff
@@ -62,7 +62,7 @@ macro_rules! binary_operator {
             type Output = BigUint<M>;
 
             fn $method(self, other: u64) -> BigUint<M> {
-                let other_handle = M::managed_type_impl().bi_new(other as i64);
+                let other_handle = M::managed_type_impl().bi_new_handle(other as i64);
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:65]: Same.
We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 4]
```diff
@@ -72,7 +72,7 @@ macro_rules! binary_operator {
...
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:75]: Same.
We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 5]
```diff
@@ -108,14 +108,14 @@ macro_rules! binary_assign_operator {
 
         impl<M: ManagedTypeApi> $trait<u32> for BigUint<M> {
             fn $method(&mut self, other: u32) {
-                let other_handle = M::managed_type_impl().bi_new(other as i64);
+                let other_handle = M::managed_type_impl().bi_new_handle(other as i64);
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:111]: Same.
We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

File: elrond-wasm/src/types/managed/big_uint_operators.rs [hunk 6]
```diff
@@ -108,14 +108,14 @@ macro_rules! binary_assign_operator {
...
                 M::managed_type_impl().$api_func(self.handle, self.handle, other_handle);
             }
         }
 
         impl<M: ManagedTypeApi> $trait<u64> for BigUint<M> {
             fn $method(&mut self, other: u64) {
-                let other_handle = M::managed_type_impl().bi_new(other as i64);
+                let other_handle = M::managed_type_impl().bi_new_handle(other as i64);
```
Comment (andrei-marinica) [inline elrond-wasm/src/types/managed/big_uint_operators.rs:118]: Same.
We need to use something based on `bigIntSetInt64` here.
One of the temporary handles can be used.

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — DISMISSED** [review]

**Review by claudiu725 — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**